### PR TITLE
Tree noIndent allows for nested Tree to align with adjacent TreeItems

### PR DESCRIPTION
**Pull request by lukelooker**
### :sparkles: Changes

- `Tree` `noIndent` prop now causes `Tree` to be indented at the same depth as adjacent `TreeItem`(s)

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [x] Check for image-snapshot changes (run `yarn image-snapshots` locally)
- [x] PR is ideally < ~400LOC

### CODE REVIEW

**Review by ElliotTheHuman — APPROVED** [review]
Code looks good! Just a few nits.

Not a blocker, but I wonder if the name `noIndent` could be confusing to someone (`noIndent` makes me think we completely remove left indent on a given `Tree`, but technically the prop lines it up with any sibling `TreeItem`s as described). 

Maybe `alignAsItem`, which isn't a great suggestion tbh? Ultimately, if you like `noIndent` I'm totally on board with keeping it!

File: packages/components/src/Tree/stories/FieldPicker.story.tsx [hunk 1]
```diff
@@ -116,7 +116,7 @@ const PickerItem = ({ children = 'Cost', truncate = false }) => {
           onClick={() => alert('Clicked on cost!')}
           onMetaEnter={() => alert("Cmd + Enter'ed on cost!")}
           selected={!!overlay}
-          icon="FieldNumber"
+          // icon="FieldNumber"
```
[the remark this answers is not in the document]
Comment (lukelooker) [reply packages/components/src/Tree/stories/FieldPicker.story.tsx]: Yes!

**Comment by lukelooker** [comment]
> Maybe alignAsItem, which isn't a great suggestion tbh? Ultimately, if you like noIndent I'm totally on board with keeping it!

Went with `branchAlign` & branchFontWeight` to unify naming :)

**Review by ElliotTheHuman — DISMISSED** [review]
LGTM!

**Review by ElliotTheHuman — APPROVED** [review]
Updated `FieldPicker` story to use new prop names. LGTM!

**Pull request merged**